Extensions/one_click_postage.js

```diff
@@ -214,237 +214,122 @@ XKit.extensions.one_click_postage = new Object({
 		});
 	},
 
-	in_blog_post: function(obj, state, retry_mode) {
-		if ($(obj).hasClass("xkit-button-working") === true) { return; }
+	in_blog_post: function($button, state, retry_mode) {
+		if ($button.hasClass("xkit-button-working")) { return; }
 
-		$(obj).removeClass("xkit-button-error");
-		$(obj).removeClass("xkit-button-done");
-		$(obj).addClass("xkit-button-working");
+		$button
+			.removeClass("xkit-button-error")
+			.removeClass("xkit-button-done")
+			.addClass("xkit-button-working");
 
-		var post_id = parseInt(XKit.iframe.single_post_id());
-		var form_key = XKit.iframe.form_key();
-		var reblog_key = XKit.iframe.reblog_button()[0].pathname.split('/')[3];
+		var request = {
+			reblog_id: parseInt(XKit.iframe.single_post_id()),
+			reblog_key: XKit.iframe.reblog_button()[0].pathname.split('/')[3]
+		};
 
-		var m_blogs = XKit.tools.get_blogs();
-		var blog_id = "";
+		XKit.interface.post.fetch(request).then(
 
-		if (!m_blogs) {
-			XKit.window.show("Error",
-				"Can't get list of current blogs - please visit dashboard first.",
-				"error",
-				'<div class="xkit-button default" id="xkit-close-message">OK</div>'
-			);
-			return;
-		} else {
-			for (var i = 0; i < m_blogs.length; i++) {
-				if (m_blogs[i] !== "") {
-					blog_id = m_blogs[i];
-					break;
+			function(data) {
+				if (data.errors === false) {
+					XKit.extensions.one_click_postage.in_blog_process(data, state, $button, request, false);
+				} else {
+					XKit.extensions.one_click_postage.show_error("OCP05-B", state);
 				}
-			}
-		}
-
-		XKit.extensions.one_click_postage.blogs_list = XKit.tools.get_blogs();
-
-		if (XKit.extensions.one_click_postage.preferences.default_blog.value !== "") {
-			if (XKit.extensions.one_click_postage.blogs_list.indexOf(XKit.extensions.one_click_postage.preferences.default_blog.value) !== -1) {
-				var channel_id = XKit.extensions.one_click_postage.preferences.default_blog.value;
-				blog_id = channel_id;
-			} else {
-				$(obj).removeClass("xkit-button-working");
-				$(obj).addClass("xkit-button-error");
-				return;
-			}
-		}
-
-		var m_object = {
-			channel_id: blog_id,
-			reblog_id: post_id,
-			reblog_key: reblog_key,
-			form_key: form_key,
-			post_type: ""
```

Comment:
did we remove this on purpose? XKit.interface.fetch_post seems to indicate it's required

Reply:
post_type: ""